src/ycc_hull/controllers/notifications/helpers_notifications_controller.py

```diff
@@ -1,28 +1,239 @@
+                "timing",
+                self._timing_changed,
+                lambda: format_helper_task_timing(self._old_task),
+            ),
+            (
+                "helpers needed",
+                self._helper_min_max_count_changed,
+                lambda: format_helper_task_min_max_helpers(self._old_task),
+            ),
+            (
+                "urgent" if self._new_task.urgent else "not urgent",
+                self._urgent_changed,
+                None,
+            ),
+            (
+                "published" if self._new_task.published else "unpublished",
+                self._published_changed,
+                None,
+            ),
+            ("captain", self._captain_changed, None),
+            ("helpers", self._helpers_changed, None),
+            ("status", self._status_changed, None),
+        ]
+
+        for label, change in diff_changes:
+            if change:
+                self._add_detailed_change(label, change["old"])
+
+        for label, flag, get_previous_value in flagged_changes:
+            if flag:
+                if get_previous_value:
+                    self._add_detailed_change(label, get_previous_value())
+                else:
+                    self.summary.append(label)
+
+    def _collect_undetected_changes(self) -> None:
+        for remaining_key in self._diff.keys():
+            self._logger.warning(
+                "Unhandled field for task update notification: %s", remaining_key
+            )
+            what = camel_case_to_words(remaining_key)
+            self.summary.append(what)
+            self.relevant_details[f"Previous {what}"] = self._diff[remaining_key]["old"]
+
 
 class HelpersNotificationsController(BaseController):
     """
     Controller for sending helper task notifications.
     """
 
+    async def on_update(
+        self,
+        old_task: HelperTaskDto,
+        new_task: HelperTaskDto,
+        diff: dict[str, DiffEntry],
+        user: User,
+    ) -> None:
+        if not CONFIG.emails_enabled(self._logger):
+            return
+
+        changes = _HelperTaskChanges(old_task, new_task, diff)
+
+        if changes.summary:
+            changes_str = f"Here is what changed: {', '.join(changes.summary)}."
+        else:
+            changes_str = "Nothing seems to have changed — but it's still worth checking in the app! 😊"
+
+        previous_values_html = ""
+        if changes.relevant_details:
+            previous_values_html = f"""
+<p>Details:</p>
+
+<table>
+    {"\n".join(f"  <tr><td>{key}</td><td>{value}</td></tr>" for key, value in changes.relevant_details.items())}
+</table>
+"""
+
+        message = (
+            _task_notification_email_to_all_participants(new_task, user, old_task)
+            .content(
+                wrap_email_html(
+                    f"""
+{_DEAR_SAILORS}
+
+<p>{user.full_name} has updated this task. 📢</p>
+
+<p>{changes_str}</p>
+
+{format_helper_task(new_task)}
+
+{previous_values_html}
+"""
```

Comment:
maybe add timestamp to the content of the mail? if things are urgent and there's some delays in the email delivery then could cause confusion.

Reply:
Thank you, great point, I think I will do it when I get a review on the email texts from the communcation expert! https://github.com/Yachting-Club-CERN/ycc-hull/issues/82